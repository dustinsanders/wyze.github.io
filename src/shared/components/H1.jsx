// @flow

import type { Style, VNode } from '../../types'
import { css } from 'glamor'
import preact from 'preact'

type Props = {
  children?: VNode,
  styles?: Style,
}

const h1: Style = css({
  fontSize: '2em',
})

const H1 = ({ children, styles }: Props): VNode => (
  <h1 {...css(h1, styles)}>{children}</h1>
)

H1.defaultProps = {
  children: null,
  styles: {},
}

export default H1
