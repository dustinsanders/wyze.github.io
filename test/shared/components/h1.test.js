// @flow

import H1 from '../../../src/shared/components/H1'
import preact from 'preact'
import snapshot from '../../helpers/snapshot'

const children = (
  // eslint-disable-next-line react/jsx-filename-extension
  <span>Hello</span>
)

const styles = {
  color: 'rebeccapurple',
}

describe('<H1 />', () => {
  it('renders', snapshot.bind(null, H1, { children }))
  it('renders styles', snapshot.bind(null, H1, { children, styles }))
})
