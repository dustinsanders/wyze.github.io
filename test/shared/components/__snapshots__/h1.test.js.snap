// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> renders 1`] = `
"<h1 data-css-4crvu7>
  <span>Hello</span>
</h1>"
`;

exports[`<H1 /> renders styles 1`] = `
"<h1 data-css-lkq5a>
  <span>Hello</span>
</h1>"
`;
